### PR TITLE
Videolisting categorylisting

### PR DESCRIPTION
**Pull request by manojsarna**
- [x] Video Listing Page Added
- [x] Category Listing Page Added
- [x] Single Video Page Added
- [x] Data from API Calls
- [x] Loader Added
 
- [x] Dark Theme Added
- [x] Mockman Integration

Deploy Link : https://incomparable-gnome-0857d1.netlify.app/

### Files To Check

- [ ] [SmashPlayer.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/videolisting-categorylisting?expand=1#diff-99ef2deb195d38e18605013290ae302e99cc57130669f7e224cbad50f3413a73)

- [ ] [/Auth.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/videolisting-categorylisting?expand=1#diff-05a2ef413bbfcc66caca4c90c620ada11233c10128abf287208bfdeddc429ba8)

- [ ] [CategoryPage.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/videolisting-categorylisting?expand=1#diff-f7960e88b3ded03175d9efed0cb732f4b489f789a400ff80b75f4c7941d888a8)

- [ ] [Home.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/videolisting-categorylisting?expand=1#diff-172462f7215de52769ba39af6ddb83fcc09fcaefa36a75651b75c68a0e43a05a)

- [ ] [VideoPage.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/videolisting-categorylisting?expand=1#diff-b75b4e4f13a411dae63fcbcd94adfddc796688b39c9769cbb8d693237d6857b3)

- [ ] [Videos.jsx](https://github.com/manojsarna/manoj-smashtube-video-lib-react/compare/videolisting-categorylisting?expand=1#diff-1a6a8a136c50222a916915a6f25d5936d8d2caba5f646f3b48b0f7bcd9f710a9)

### Ignore Rest of the Files

image
image
image
image

### CODE REVIEW

File: src/pages/auth/Auth.jsx [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
+import { Login, SignUp } from "./components";
 
 export function Auth() {
   useDocTitle("Auth - SmashCart - Manoj Sarna");
-  return <main className="auth">Auth Page</main>;
+  const [authVal, setAuthVal] = useState(true);
+
+  return (
+    <main className="sm-main-auth">
+      {authVal ? (
```
Comment (dev-ashishsethi) [inline src/pages/auth/Auth.jsx:12]: you can use context for authentication

File: src/pages/video-page/VideoPage.jsx [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
+    (async () => {
+      try {
+        setLoading(true);
+        const response = await axios.get(`/api/videos/${videoId}`);
+        if (response.status === 200) {
+          setVideoDetails(response.data.video);
+          setLoading(false);
+        }
+      } catch (error) {
+        console.log(error);
```
Comment (dev-ashishsethi) [inline src/pages/video-page/VideoPage.jsx:32]: use console.error for errors

File: src/pages/home/Home.jsx [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
+  const [videos, setVideos] = useState([]);
+  useEffect(() => {
+    (async function () {
+      try {
+        const { data } = await axios.get("/api/videos");
+        setVideos(data.videos);
+      } catch (error) {
+        console.log("error", error);
+      }
+    })();
```
Comment (dev-ashishsethi) [inline src/pages/home/Home.jsx:30]: good use of IIFE

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

File: src/components/smash-player/SmashPlayer.jsx [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import "./smashplayer.css";
-export function SmashPlayer({ videoId }) {
+export function SmashPlayer({ videoId, videoDetails }) {
   const videoEmbed = "https://www.youtube.com/embed/";
-  const videoSrc = `${videoEmbed}${videoId}`;
+  const videoSrc = `${videoEmbed}${videoId}?autoplay=1`;
```
Comment (rahulyadav139) [inline src/components/smash-player/SmashPlayer.jsx:4]: Great you have implemented auto play for youtube videos. Great work.

**Review by rahulyadav139 — APPROVED** [review]

**Pull request merged**